src/ComposerGpgVerify/Verify.php

```diff
@@ -0,0 +1,213 @@
+                    escapeshellarg($vendorDir->getRealPath() . '/.git')
+                ),
+                $output,
+                $signed
+            );
+
+            $signed = $signed || ! self::verifySignatureValidationHasNoWarnings($output);
+
+            if ($signed) {
+                // again, moronic language.
+                $tags = [];
+
+                exec(
+                    sprintf(
+                        'git --git-dir %s tag --points-at HEAD 2>&1',
+                        escapeshellarg($vendorDir->getRealPath() . '/.git')
+                    ),
+                    $tags
+                );
+
+                // go through all found tags, see if at least one is signed
+                foreach (array_filter($tags) as $tag) {
+                    exec(
+                        sprintf(
+                            'git --git-dir %s tag -v %s 2>&1',
+                            escapeshellarg($vendorDir->getRealPath() . '/.git'),
+                            escapeshellarg($tag)
+                        ),
+                        $tagSignatureOutput,
+                        $signed
+                    );
+
+                    $signed = $signed || ! self::verifySignatureValidationHasNoWarnings($tagSignatureOutput);
+
+                    if (! $signed) {
+                        $output = array_merge($output, $tagSignatureOutput);
+
+                        break;
+                    }
+                }
+            }
+
+            $packages[$packageName] = [
+                'git'       => 'dunno',
+                'signed'    => (bool) $output,
+                'signature' => implode("\n", $output),
+                'verified'  => ! $signed,
+            ];
+        }
+
+        // @TODO configuration WILL arrive, but for now I'm too lazy, so GTFO
+//        $settings = $composer->getPackage()->getExtra()['composer-gpg-verify'];
+//
+//        $allowedUnsigned  = $settings['allow-unsigned'];
+//        $allowedUntrusted = $settings['allow-untrusted'];
+
+        $allowedUnsigned  = [];
+        $allowedUntrusted = [];
+
+
+        $unSigned = array_keys(array_filter(
+            $packages,
+            function (array $package) : bool {
+                return ! $package['signed'];
+            }
+        ));
+
+        $unVerified = array_keys(array_filter(
+            $packages,
+            function (array $package) : bool {
+                return ! $package['verified'];
+            }
+        ));
+
+        $requiringSignature    = array_diff($unSigned, $allowedUnsigned);
+        $requiringVerification = array_diff($unVerified, $allowedUntrusted);
+
+        $escapes = array_values(array_unique(array_merge($requiringSignature, $requiringVerification)));
+
+        if (! $escapes) {
+            return;
+        }
+
+        // @TODO report un-trusted signatures emails, names, identifiers (full length, if possible)
+
+        throw new \RuntimeException(sprintf(
```

Comment:
AFAIK, the package of the plugin is registered in a temporary autoloader during the run of composer (to have access to the plugin), so it may be possible to use a custom exception.